### PR TITLE
tag search works like OR filter, not AND

### PR DESCRIPTION
**Pull request by IdiosApps**
- Used TDD red/green method
     - Decode two packages (tagged with `"A"` and `"B"` respectively)
     - Send them to tagsSearch function, with a tag search of `["A","B"]`
     - Now tagsSearch returns both packages. Previously it returned none.

### CODE REVIEW

File: test/api/search.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      "version": "1.2.3",
+      "archive_url": "",
+      "archive_sha256_url": ""
+    }
+  };
+
+  test("selecting multiple tags combines all possible results", () => {
+    const packagesJson = [
+      generatePackageWithTags(["a"]),
+      generatePackageWithTags(["b"]),
```
Comment (arabello) [inline test/api/search.test.ts:27]: It's already ok as the current use case is simple, but I would add a third package to the test so we are sure that `tagsSearch` is actually filtering, even with future changes. What do you think?
```suggestion
      generatePackageWithTags(["a"]),
      generatePackageWithTags(["b"]),
      generatePackageWithTags(["c"])
```

**Review by arabello — COMMENTED** [review]
LGTM 🚀 

I just have a minor suggestion, then we can merge it

File: test/api/search.test.ts (hunk 1 above)
Comment (IdiosApps) [reply test/api/search.test.ts:27]: I'm thinking we take the idea a step further, like:
- each package can have multiple tags, a random sample from A...F 
- we have 3 packages like you suggest 
- the tags to be filtered is all of A...F 
And so we still expect all packages to show

Multiple tags could mean we can also ensure the results are sorted by best match

Comment (arabello) [reply test/api/search.test.ts:27]: - Having multiple tags on each package is a good idea
- I would avoid to have the filtered array length equals to the starting one: it does not test filtering out wrong packages
- I would avoid randomly generated tags and the ones to be filtered should be a subset of the total present in all packages

In my mind this assure more robustness :) we can address the sorted matching in another PR

**Pull request merged**